src/abstractdataframe/selection.jl

```diff
@@ -160,7 +178,9 @@ function select_transform!(nc::Pair{<:Union{Int, AbstractVector{Int}},
             newdf[!, newname] = res
         end
     else
-        newdf[!, newname] = [res]
+        res_unwrap = res isa Union{AbstractArray{<:Any, 0}, Ref} ? res[] : res
+        # disallow squashing a scalar to 0 rows
+        newdf[!, newname] = fill(res_unwrap, max(1, nrow(newdf)))
```

Comment:
```suggestion
        newdf[!, newname] = fill!(Tables.allocatecolumn(typeof(res_unwrap), max(1, nrow(newdf))), res_unwrap)
```

Reply:
maybe `fill` should be overloaded in CategoricalArrays.jl. Its contract in Base does not promise a standard array?

Reply:
Yes, probably. I've filed https://github.com/JuliaData/CategoricalArrays.jl/issues/257.

Reply:
Can you add tests with `CategoricalArray` BTW?

Reply:
Tests added - all was good, but still I discovered that there are many possible cases we should test against.